apiary.apib

```diff
@@ -4,7 +4,107 @@ FORMAT: 1A
 
 The API for interacting with IPFS nodes.
 
-# Group add
+# Group add [POST /add{?arg}{&recursive,quiet,silent,progress,trickle,only%2Dhash,wrap%2Dwith%2Ddirectory,hidden,chunker,pin}]
+Add an object to IPFS.
+
+Note that directories are added recursively, to form the ipfs
+MerkleDAG. A smarter partial add with a staging area (like git)
+remains to be implemented.
+
+#### curl
+
+    curl -F "file=@test/" -r http://localhost:5001/api/v0/add
+
++ Parameters
+    + arg (string, required) - The path to a file to be added to IPFS
+    + recursive (boolean, optional) - Add directory paths recursively
+    + quiet (boolean, optional) - Write minimal output
+    + silent (boolean, optional) - Write no output
+    + progress (boolean, optional) - Stream progress data
+    + trickle (boolean, optional) - Use trickle-dag format for dag generation
+    + only-hash (boolean, optional) - Only chunk and hash - do not write to disk
+    + wrap-with-directory (boolean, optional) - Wrap files with a directory object
+    + hidden (boolean, optional) - Include files that are hidden
+    + chunker (boolean, optional) - chunking algorithm to use
+    + pin (boolean, optional) - Pin this object when adding.  Default true
+
++ Request Single File (application/json)
+
+    #### Curl
+
+        curl -F "file=test" http://localhost:5001/api/v0/add
+
+    + Body
+
+        ```
+        curl -F "file=test" http://localhost:5001/api/v0/add
+        ```
+
++ Response 200 (application/json)
+
+    + Headers
+
+        ```
+        HTTP/1.1 100 Continue
+
+        HTTP/1.1 200 OK
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Chunked-Output: 1
+        Date: Fri, 08 Jan 2016 14:48:18 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes
+        + Name (string)
+        + Hash (Multihash)
+
+    + Body
+
+        ```
+        {
+            "Name":"test",
+            "Hash":"QmePw8gVcBMb8x6kAep6aMBAX23hCSk6iZW3i9VKkiFhu1"
+        }
+        ```
+
++ Request Recursive files (text/plain)
```

Comment:
Just pinged @whyrusleeping , we actually use `multipart/form-data` for this, but now I'm not sure what should be the correct one.